### PR TITLE
parametes for spark, hive, hcat

### PR DESCRIPTION
**Pull request by axier**
added parameters to customize home directories

### CODE REVIEW

File: templates/service/unit-kylin.erb [hunk 1]
```diff
@@ -7,9 +7,9 @@ Type=forking
 User=<%= scope.lookupvar('kylin::kylin_user') %>
 Group=<%= scope.lookupvar('kylin::kylin_group') %>
 Environment='JAVA_HOME=<%= scope.lookupvar('kylin::java_home') %>'
-Environment='SPARK_HOME=/usr/hdp/2.5.3.0-37/spark'
-Environment='HIVE_CONF=/usr/hdp/2.5.3.0-37/hive/conf'
-Environment='HCAT_HOME=/usr/hdp/2.5.3.0-37/hive-hcatalog'
+Environment='SPARK_HOME=<%= scope.lookupvar('kylin::spark_home') %>'
+Environment='HIVE_CONF=<%= scope.lookupvar('kylin::hive_home') %>/conf'
+Environment='HCAT_HOME=<%= scope.lookupvar('kylin::hcat_home') %>'
```
Comment (ElvenSpellmaker) [inline templates/service/unit-kylin.erb:12]: We shouldn't really be using `scope.lookupvar` but as it's used elsewhere in the file, I suppose we can let it slide.

File: manifests/params.pp [hunk 2]
```diff
@@ -19,6 +19,9 @@
   $install_java               = false
   $java_version               = '8'
   $java_home                  = '/usr/jdk64/jdk1.8.0_77'
+  $spark_home                 = '/usr/hdp/2.5.3.0-37/spark'
+  $hive_home                  = '/usr/hdp/2.5.3.0-37/hive'
+  $hcat_home                  = '/usr/hdp/2.5.3.0-37/hive-hcatalog'
```
Comment (ElvenSpellmaker) [inline manifests/params.pp:24]: What are those version numbers hardcoded there, do they come from another variable?

Comment (axier) [reply manifests/params.pp:24]: It has ambari version in the path but in the new cluster the path changes a little bit and for that reason it doesn't have sense to use the ambari version as a parameter.

**Comment by juan-martinez-herrero** [comment]
LGTM +1

File: manifests/params.pp (hunk 2 above)
Comment (aasier) [reply manifests/params.pp:24]: After the explanation of Axier, it looks good for me...seeing that puppet will dead after K8s in the following month.
LGTM

**Pull request merged**